[2023-10-03T14:50:27.013+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: Reddit_Post_Dag.extract_post_task manual__2023-10-03T14:48:59.438533+00:00 [queued]>
[2023-10-03T14:50:27.041+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: Reddit_Post_Dag.extract_post_task manual__2023-10-03T14:48:59.438533+00:00 [queued]>
[2023-10-03T14:50:27.043+0000] {taskinstance.py:1361} INFO - Starting attempt 2 of 2
[2023-10-03T14:50:27.074+0000] {taskinstance.py:1382} INFO - Executing <Task(PythonOperator): extract_post_task> on 2023-10-03 14:48:59.438533+00:00
[2023-10-03T14:50:27.086+0000] {standard_task_runner.py:57} INFO - Started process 188 to run task
[2023-10-03T14:50:27.092+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'Reddit_Post_Dag', 'extract_post_task', 'manual__2023-10-03T14:48:59.438533+00:00', '--job-id', '6', '--raw', '--subdir', 'DAGS_FOLDER/reddit_post_dag.py', '--cfg-path', '/tmp/tmp4_54c2v_']
[2023-10-03T14:50:27.100+0000] {standard_task_runner.py:85} INFO - Job 6: Subtask extract_post_task
[2023-10-03T14:50:27.189+0000] {task_command.py:415} INFO - Running <TaskInstance: Reddit_Post_Dag.extract_post_task manual__2023-10-03T14:48:59.438533+00:00 [running]> on host a82955aebbba
[2023-10-03T14:50:27.298+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='Reddit_Post_Dag' AIRFLOW_CTX_TASK_ID='extract_post_task' AIRFLOW_CTX_EXECUTION_DATE='2023-10-03T14:48:59.438533+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2023-10-03T14:48:59.438533+00:00'
[2023-10-03T14:50:53.258+0000] {logging_mixin.py:151} INFO -                                                  title  ...  subreddit
0                              THEY WON THE UNION VOTE  ...  starbucks
1                   60 Frappuccino‚Äôs with no tip‚Ä¶ $279  ...  starbucks
2               I'm not allowed to make signs anymore.  ...  starbucks
3                          What did you get fired for?  ...  starbucks
4                          An inspiration to us all üôåüèª  ...  starbucks
..                                                 ...  ...        ...
992                                    it be like that  ...  starbucks
993  Not sure which makes me happier, our raises or...  ...  starbucks
994  What's your favorite moment of instant karma w...  ...  starbucks
995        These modifications are getting ridiculousüôÑ  ...  starbucks
996   My manager made the moment right in the best way  ...  starbucks

[997 rows x 9 columns]
[2023-10-03T14:50:53.265+0000] {taskinstance.py:1943} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 192, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 209, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/plugins/reddit_data_extractor.py", line 51, in extract_post_data
    topics_data.to_csv(csv_file_path, index=False)
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/core/generic.py", line 3772, in to_csv
    return DataFrameRenderer(formatter).to_csv(
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/formats/format.py", line 1186, in to_csv
    csv_formatter.save()
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/formats/csvs.py", line 240, in save
    with get_handle(
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/common.py", line 737, in get_handle
    check_parent_directory(str(handle))
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/common.py", line 600, in check_parent_directory
    raise OSError(rf"Cannot save file into a non-existent directory: '{parent}'")
OSError: Cannot save file into a non-existent directory: 'data/raw'
[2023-10-03T14:50:53.292+0000] {taskinstance.py:1400} INFO - Marking task as FAILED. dag_id=Reddit_Post_Dag, task_id=extract_post_task, execution_date=20231003T144859, start_date=20231003T145027, end_date=20231003T145053
[2023-10-03T14:50:53.316+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 6 for task extract_post_task (Cannot save file into a non-existent directory: 'data/raw'; 188)
[2023-10-03T14:50:53.365+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2023-10-03T14:50:53.398+0000] {taskinstance.py:2784} INFO - 0 downstream tasks scheduled from follow-on schedule check
